punishments.js

```diff
@@ -726,6 +726,23 @@ Punishments.roomUnblacklist = function (room, userid) {
 	return Punishments.roomUnpunish(room, userid, 'BLACKLIST');
 };
 
+Punishments.roomUnblacklistAll = function (room) {
+	const roombans = Punishments.roomUserids.get(room.id);
+	if (!roombans) return false;
+
+	let success = false;
+
+	roombans.forEach((punishment, userid) => {
+		if (punishment[0] === 'BLACKLIST') {
+			Punishments.roomUnblacklist(room, userid);
+			success = true;
+		}
+	});
+	if (!success) return false;
+	Punishments.savePunishments();
+	return success;
```

Comment:
Well, it's not completely revertible, unless you log IPs/alts, and even then there's not a good way to add IPs/alts to a blacklist.